src/server/project.ts

```diff
@@ -202,7 +202,93 @@ namespace ts.server {
 
             this.setInternalCompilerOptionsForEmittingJsFiles();
 
-            this.lsHost = new LSHost(this.projectService.host, this, this.projectService.cancellationToken);
+            let proxyHost:ServerHost = (<any>Object).assign({}, this.projectService.host);
+            if (projectKind === ProjectKind.Configured && hasExplicitListOfFiles) {

```

Comment:
It's a constructor parameter to the Project from the ProjectService that creates it.